classes/Tools.php

```diff
@@ -88,29 +93,33 @@ public static function resetRequest()
      *
      * @return bool|string Password
      */
-    public static function passwdGen($length = 8, $flag = 'ALPHANUMERIC')
+    public static function passwdGen($length = 8, $flag = null)
     {
         $length = (int) $length;
 
         if ($length <= 0) {
             return false;
         }
 
+        if (null === $flag) {
+            $flag = static::PASSWORDGEN_FLAG_ALPHANUMERIC;
+        }
+
         switch ($flag) {
-            case 'NUMERIC':
+            case static::PASSWORDGEN_FLAG_NUMERIC:
                 $str = '0123456789';
 
                 break;
-            case 'NO_NUMERIC':
+            case static::PASSWORDGEN_FLAG_NO_NUMERIC:
                 $str = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ';
 
                 break;
-            case 'RANDOM':
+            case static::PASSWORDGEN_FLAG_RANDOM:
```

Comment:
Why not abcdefghijkmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ ?

Reply:
It is outside the scope of this PR. Can you please open a new PR ?